lib/puppet/type/pulp_role.rb

```diff
@@ -0,0 +1,28 @@
+
+Puppet::Type.newtype(:pulp_role) do
+  @doc = <<-EOT
+    doc
+  EOT
+
+  autorequire(:file) do
+    ['/etc/pulp/admin/admin.conf']
+  end
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
+  newparam(:name, :namevar => true) do
+    desc "role-id: uniquely identifies the role"
+  end
+
+  newproperty(:users, :array_matching => :all) do
+    desc "users to add to this role"
+  end
+
+  newproperty(:permissions) do
```

Comment:
No, it's a hash which contains arrays. It's thus handled in a much more verbose way (I couldn't make it shorter) in lib/puppet/provider/pulp_role/api.rb